pkg/common/common.go

```diff
@@ -163,7 +163,7 @@ func CreateHwEvent(pubSubID, eventType string, data hwevent.Data) (hwevent.Event
 	event := v1hwevent.CloudNativeEvent()
 	event.ID = pubSubID
 	event.Type = eventType
-	event.SetTime(types.Timestamp{Time: time.Now().UTC()}.Time)
+	event.SetTime(t)
```

Comment:
time should be time created, why passing the time? 
In publishEvent function you are creating the time not using time received 
`// Use time received to calculate lapse
	timeReceived := types.Timestamp{Time: time.Now().UTC()}.Time`

Also you can set the time after you received the  event object

Reply:
updated to use timestamp from Redfish event

Reply:
instead of passing the time why don't you set the time after you received the event object ?

Reply:
https://github.com/redhat-cne/cloud-event-proxy/blob/42dc03d5cd2416e27772bd9f60e803ac0a7db11f/plugins/hw_event/hw_event_plugin.go#L108
here create the time after you got the event object . That way it  will  be more accurate

Reply:
done. Set the event.Time when received Redfish event from webhook.